### ISSUE TITLE
go.vim syntax file does not work standalone

### ISSUE THREAD

**Issue by mikeweilgart** [fatih/vim-go#2285]
I downloaded https://raw.githubusercontent.com/fatih/vim-go/v1.20/syntax/go.vim as for now I just want syntax highlighting; I don't want to use all the other features.  (Possibly I'll use them in the future.)

I expected to get syntax highlighting of my go source files.

When I ran `:set syntax=go`, I got a large number of error messages:

```
Error detected while processing /Users/Mike/term/repos/local/configs/vim/syntax/go.vim:
line   64:
E117: Unknown function: go#config#FoldEnable
E15: Invalid expression: go#config#FoldEnable('comment')
line   74:
E117: Unknown function: go#config#HighlightGenerateTags
E15: Invalid expression: go#config#HighlightGenerateTags()
line   99:
E117: Unknown function: go#config#HighlightStringSpellcheck
E15: Invalid expression: go#config#HighlightStringSpellcheck()
line  107:
E117: Unknown function: go#config#HighlightFormatStrings
E15: Invalid expression: go#config#HighlightFormatStrings()
line  135:
E117: Unknown function: go#config#FoldEnable
E15: Invalid expression: go#config#FoldEnable('block')
line  142:
E117: Unknown function: go#config#FoldEnable
E15: Invalid expression: go#config#FoldEnable('import')
line  149:
E117: Unknown function: go#config#FoldEnable
E15: Invalid expression: go#config#FoldEnable('varconst')
line  192:
E117: Unknown function: go#config#HighlightArrayWhitespaceError
E15: Invalid expression: go#config#HighlightArrayWhitespaceError()
line  197:
E117: Unknown function: go#config#HighlightChanWhitespaceError
E15: Invalid expression: go#config#HighlightChanWhitespaceError()
line  215:
E117: Unknown function: go#config#HighlightExtraTypes
E15: Invalid expression: go#config#HighlightExtraTypes()
line  224:
E117: Unknown function: go#config#HighlightSpaceTabError
E15: Invalid expression: go#config#HighlightSpaceTabError()
...
```

After I finish looking at the errors (by pressing `q`, I do get syntax highlighting as I had hoped for.

It would be nice if the syntax file included conditional guards around the function references, so that those portions only applied if the functions were defined.

In other words, I think it would be good to make the syntax highlighting file work as a standalone file.

**Comment by bhcleek** [comment]
Duplicate of #2250
